mwtext/filter_functions/all_pages_and_revisions.py

```diff
@@ -1,2 +1,4 @@
-def include(page, revision):
-    return True
+def build_include():
```

Comment:
Why have a function we execute in order to create the include function?  Is there a scenario where we import one of these files and we *don't* want to use the include function?

Reply:
oh right! I thought this would be required after the changes made to wikidata_items_with_wikipedia_sitelinks. But looks like it isn't necessary.

Reply:
for wikidata_items_with_wikipedia_sitelinks the outer function was used so that the file is read only once. I thought that was what you suggested. Did I misunderstand anything?

Reply:
You can structure the module like this to make sure the file is read only once:

```
exclude_subclasses = ... load from the file ...

def include(...):
   ...

   if qid in exclude_subclasses:
        return False
   
   ...
```

That way the data will be loaded from the file when wikidata_items_with_wikipedia_sitelinks is imported and the include() function will keep re-using the same `exclude_subclasses` variable.